csm_web/scheduler/management/commands/checkintegrity.py

```diff
@@ -44,124 +26,31 @@ def _err(self, msg):
-            if section.course.name not in courses:
-                self._err("Section {} has bad course".format(section))
-            if section.course.name == "EE16A":
-                if section.default_spacetime.duration < dt.timedelta(
-                    hours=1, minutes=30
-                ):
-                    self._err("16A section {} has bad duration".format(section))
-            if not section.capacity:
-                self._err("Section {} lacks capacity".format(section))
-            if section.current_student_count > section.capacity:
-                self._err(
-                    "Section {} is overbooked ({} out of {})".format(
-                        section, section.current_student_count, section.capacity
-                    )
-                )
-            if not section.default_spacetime:
-                self._err("Section {} lacks spacetime".format(section))
-            st = section.default_spacetime
-            obj = (
-                st.start_time,
-                st.location,
-                st.day_of_week,
-                section.mentor.user.email,
-            )
-            if obj in seen_spacetimes:
-                self._err(
-                    "Duplicate spacetime for section {} by mentor {}".format(
-                        section, section.mentor
-                    )
-                )
-            seen_spacetimes.add(obj)
-            if not section.mentor:
-                self._err("Section {} lacks mentor".format(section))
-            mentor_role = section.mentor.role
-            if section.mentor.role == Profile.STUDENT:
-                self._err(
-                    "Somehow, mentor {} of section {} has role student".format(
-                        section.mentor, section
-                    )
-                )
-
-    def _check_profile_integrities(self):
-        for profile in Profile.objects.all():
-            if not profile.user:
-                self._err("Somehow, profile {} has no user".format(profile))
-            if profile.role not in Profile.ROLE_MAP:
-                self._err("Profile {} has bad role".format(profile))
-            if profile.role == Profile.STUDENT and not profile.section:
-                self._err("Student profile {} has no section".format(profile))
-            if profile.role == Profile.STUDENT and profile.active:
-                if profile.leader != profile.section.leader:
-                    self._err(
-                        "Student profile {} has inconsistent leader".format(profile)
-                    )
-
-    def _check_attendance_integrities(self):
-        qs = Attendance.objects.exclude(section=F("attendee__section"))
-        if qs.count() > 0:
-            self._err(
-                "Found {} attendances where section does not match profile".format(
-                    qs.count()
-                )
+            if section.mentor is None:
+                self._err(f"Section {section} has no mentor")
```

Comment:
This is probably just me not being as familiar with the management command framework, but is `self._err` supposed to be defined somewhere?

Reply:
I defined it on the class further above; it's just a convenience method for all errors to get printed out at the very end